src/nemos/basis.py

```diff
@@ -58,15 +60,52 @@ def wrapper(self: Basis, *xi: ArrayLike, **kwargs):
     return wrapper
 
 
-def min_max_rescale_samples(sample_pts: NDArray) -> NDArray:
-    """Rescale samples to [0,1]."""
-    if np.any(sample_pts < 0) or np.any(sample_pts > 1):
-        sample_pts -= np.min(sample_pts)
-        sample_pts /= np.max(sample_pts)
-        warnings.warn(
-            "Rescaling sample points for RaisedCosine basis to [0,1]!", UserWarning
-        )
-    return sample_pts
+def min_max_rescale_samples(
+    sample_pts: NDArray,
+    bounds: Optional[Tuple[float, float]] = None,
+) -> Tuple[NDArray, float]:
+    """Rescale samples to [0,1].
+
+    Parameters
+    ----------
+    sample_pts:
+        The original samples.
+    bounds:
+        Sample bounds. `bounds[0]` abd `bounds[1]` are mapped to 0/1 respectively.
+        Default are `min(sample_pts), max(sample_pts)`.
+
+    Raises
+    ------
+    ValueError
+        If all the samples contain invalid entries (either NaN or Inf).
+        This may happen if `max(sample) < bounds[0]` or `min(sample) >  bounds[1]`.
+
+    Warns
+    -----
+    UserWarning
+        If more than 90% of the sample points contain NaNs or Infs.
+    """
+    sample_pts = sample_pts.astype(float)
+    vmin = np.nanmin(sample_pts) if bounds is None else bounds[0]
+    vmax = np.nanmax(sample_pts) if bounds is None else bounds[1]
+    if vmin and vmax and vmax <= vmin:
+        raise ValueError("Invalid value range. `vmax` must be larger then `vmin`!")
```

Comment:
```suggestion
    if vmin and vmax and vmax <= vmin:
        raise ValueError("Invalid value range. `bounds[1]` must be larger then `bounds[0]`!")
```

since name of args have changed